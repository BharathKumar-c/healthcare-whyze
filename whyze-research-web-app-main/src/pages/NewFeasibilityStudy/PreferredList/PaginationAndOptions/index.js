import React from 'react';
import PropTypes from 'prop-types';
import { Divider } from 'antd';
import {
  ChangeTo,
  Contact,
  Delete,
  Initiated,
  LeftArrow,
  RightArrow,
} from '../../../../assets';
import { Checkbox, Text } from '../../../../components/basic';

export default function PaginationAndOptions(props) {
  const {
    tab,
    totalPage,
    currentPage,
    noOfSelectedSites,
    isCheckedAll,
    isShowOptions,
    setCurrentPage,
    handleDelete,
    handleCheckAllData,
    onClickCheckedPreferredSitesData,
    onClickCheckedContactedData,
    onClickCheckedInitiatedData,
    onTabChange,
  } = props;

  return (
    <div className="preferred-list-pagination-container">
      <div
        className={
          isShowOptions
            ? 'preferred-list-alignment'
            : 'preferred-list-checkbox-alignment'
        }
      >
        <div className="preferred-list-alignment">
          <Checkbox
            className="preferred-list-checkbox"
            name={'sites'}
            lable={`${noOfSelectedSites} Sites`}
            value={isCheckedAll}
            handleChecked={handleCheckAllData}
          />
          <Divider
            className={
              isShowOptions
                ? 'preferred-list-divider'
                : 'preferred-list-divider-hide'
            }
            type="vertical"
          />
        </div>
      </div>
      <div
        className={
          isShowOptions
            ? 'preferred-list-options'
            : 'preferred-list-options-hide'
        }
      >
        <div
          className="preferred-list-option-alignment"
          onClick={() => handleDelete()}
        >
          <div className="preferred-list-option">
            <div className="preferred-list-icon">
              <Delete />
            </div>
            <Text className="preferred-list-text" text="Delete" />
          </div>
        </div>
        {tab !== 'preferredSites' && (
          <div
            className="preferred-list-option-alignment"
            onClick={() => {
              if (noOfSelectedSites > 0) {
                onClickCheckedPreferredSitesData();
                onTabChange('preferredSites');
              }
            }}
          >
            <div className="preferred-list-option">
              <div className="preferred-list-icon">
                <ChangeTo />
              </div>
              <Text
                className="preferred-list-text"
                text="Move to preferred sites"
              />
            </div>
          </div>
        )}
        {tab !== 'contacted' && (
          <div
            className="preferred-list-option-alignment"
            onClick={() => {
              if (noOfSelectedSites > 0) {
                onClickCheckedContactedData();
                onTabChange('sitesContacted');
              }
            }}
          >
            <div className="preferred-list-option">
              <div className="preferred-list-icon">
                <Contact />
              </div>
              <Text className="preferred-list-text" text="Mark contacted" />
            </div>
          </div>
        )}
        {tab !== 'initiated' && (
          <div
            className="preferred-list-option-alignment"
            onClick={() => {
              if (noOfSelectedSites > 0) {
                onClickCheckedInitiatedData();
                onTabChange('sitesInitiated');
              }
            }}
          >
            <div className="preferred-list-option">
              <div className="preferred-list-icon">
                <Initiated />
              </div>
              <Text className="preferred-list-text" text="Mark initiated" />
            </div>
          </div>
        )}
      </div>
      <div className="preferred-list-alignment">
        <Text
          className="preferred-list-pagination-text"
          text={`${currentPage}-${totalPage} of ${totalPage}`}
        />

        <div className="preferred-list-arrow-icon">
          <LeftArrow
            className="preferred-list-arrow-left"
            onClick={() =>
              currentPage !== 1
                ? setCurrentPage((previousPage) => previousPage - 1)
                : null
            }
          />
          <RightArrow
            className="preferred-list-arrow-right"
            onClick={() =>
              currentPage !== totalPage
                ? setCurrentPage((previousPage) => previousPage + 1)
                : null
            }
          />
        </div>
      </div>
    </div>
  );
}

PaginationAndOptions.propTypes = {
  tab: PropTypes.string,
  totalPage: PropTypes.number,
  currentPage: PropTypes.number,
  noOfSelectedSites: PropTypes.number,
  isCheckedAll: PropTypes.bool,
  isShowOptions: PropTypes.bool,
  setCurrentPage: PropTypes.func,
  handleCheckAllData: PropTypes.func,
  handleDelete: PropTypes.func,
  onClickCheckedPreferredSitesData: PropTypes.func,
  onClickCheckedContactedData: PropTypes.func,
  onClickCheckedInitiatedData: PropTypes.func,
  onTabChange: PropTypes.func,
};

PaginationAndOptions.defaultProps = {
  tab: 'preferredSites',
  totalPage: 0,
  currentPage: 0,
  noOfSelectedSites: 0,
  isCheckedAll: false,
  isShowOptions: false,
  setCurrentPage: () => {},
  handleDelete: () => {},
  handleCheckAllData: () => {},
  onClickCheckedPreferredSitesData: () => {},
  onClickCheckedContactedData: () => {},
  onClickCheckedInitiatedData: () => {},
  onTabChange: () => {},
};
