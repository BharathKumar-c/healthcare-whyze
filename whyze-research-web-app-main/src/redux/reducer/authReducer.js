import { createSlice } from '@reduxjs/toolkit';
import jwt_decode from 'jwt-decode';

const initialState = {
  isLoggedIn: loginCheck(),
  accessToken: null,
  refreshToken: null,
  isLoading: false,
  isDisableForgot: false,
  message: '',
  userDetails: {},
  profileCount: 0,
};

function loginCheck() {
  const credentials = JSON.parse(localStorage.getItem('credentials'));
  if (credentials?.access_token) {
    const { exp } = jwt_decode(credentials.access_token);
    if (Date.now() <= exp * 1000) {
      return true;
    }
  }
  return false;
}

const authSlice = createSlice({
  name: 'authAction',
  initialState: initialState,
  reducers: {
    setLoginStatus: (state, { payload }) => {
      state.isLoggedIn = payload;
    },
    setAccessToken: (state, { payload }) => {
      state.accessToken = payload;
    },
    setRefreshToken: (state, { payload }) => {
      state.refreshToken = payload;
    },
    setLoading: (state, { payload }) => {
      state.isLoading = payload;
    },
    setDisableForgot: (state, { payload }) => {
      state.isDisableForgot = payload;
    },
    setResetMessage: (state, { payload }) => {
      state.message = payload;
    },
    setUserDetails: (state, { payload }) => {
      state.userDetails = payload;
    },
    setProfileCount: (state) => {
      state.profileCount = state.profileCount + 1;
    },
  },
});

export const {
  setLoginStatus,
  setAccessToken,
  setRefreshToken,
  setLoading,
  setDisableForgot,
  setResetMessage,
  setUserDetails,
  setProfileCount,
} = authSlice.actions;

export default authSlice.reducer;
