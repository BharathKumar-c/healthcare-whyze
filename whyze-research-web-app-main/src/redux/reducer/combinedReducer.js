import { combineReducers } from 'redux';
import feasibilityStudy from './feasibilityStudyReducer';
import project from './projectsReducer';
import auth from './authReducer';

export default combineReducers({
  feasibilityStudyReducer: feasibilityStudy,
  authReducer: auth,
  projectReducer: project,
});
