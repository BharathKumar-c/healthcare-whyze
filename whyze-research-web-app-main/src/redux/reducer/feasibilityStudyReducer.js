import { createSlice } from '@reduxjs/toolkit';
import { v4 as uuid } from 'uuid';

export const initialState = {
  segments: [
    {
      name: 'Segment-1',
      isDone: false,
      activeTab: {
        name: 'feasibility',
        index: 0,
      },
      tabs: [
        {
          name: 'feasibility',
          title: 'Feasibility',
          isActive: true,
          position: 'left',
        },
      ],
      feasibilityStudy: [
        {
          name: 'Indication',
          case: 'Indication',
          condition: 'Inclusion',
          caseName: '',
          child: [],
        },
        {
          name: 'Demographics',
          case: 'Demographics',
          condition: 'Inclusion',
          caseName: '',
          child: [
            {
              id: uuid().slice(0, 8),
              condition: 'Gender',
              field: 'gender',
              gender: '',
            },
            {
              id: uuid().slice(0, 8),
              condition: 'Age range',
              field: 'age',
              ageRange: {
                min: 0,
                max: 0,
              },
            },
          ],
        },
      ],
      country: [],
      location: [],
      selectedCountries: [],
      totalConnectedNotConnected: {},
      sites: [],
      sitesContacted: [],
      sitesInitiated: [],
      preferredSites: [],
      patientCount: {
        feasibility: 0,
        location: 0,
      },
      siteCount: {
        feasibility: 0,
        location: 0,
      },
    },
  ],
  cardValue: {
    case: 'Indication',
    condition: '',
    index: 0,
  },
  activeSegment: {
    name: '',
    index: 0,
  },
};

const feasibilityStudySlice = createSlice({
  name: 'feasibilityStudyAction',
  initialState: initialState,
  reducers: {
    setCardDetails: (state, { payload }) => {
      state.cardValue = payload;
    },
    setSegments: (state, { payload }) => {
      state.segments = payload;
    },
    setActiveSegment: (state, { payload }) => {
      state.activeSegment = payload;
    },
    resetSegments: (state) => {
      state.segments = initialState.segments;
      state.cardValue = {
        case: 'Indication',
        condition: '',
        index: 0,
      };
      state.activeSegment = {
        name: '',
        index: 0,
      };
    },
  },
});

export const { setCardDetails, setSegments, setActiveSegment, resetSegments } =
  feasibilityStudySlice.actions;

export default feasibilityStudySlice.reducer;
