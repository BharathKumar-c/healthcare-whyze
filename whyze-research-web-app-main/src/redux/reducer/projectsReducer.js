import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  projects: [],
  projectId: '',
  clinicalTrial: '',
  sponsor: '',
  description: '',
  measurement: 'Metric',
  standards: '',
  segmentId: [],
};

const projectSlice = createSlice({
  name: 'projectAction',
  initialState: initialState,
  reducers: {
    setProjects: (state, { payload }) => {
      state.projects = payload;
    },
    setProjectId: (state, { payload }) => {
      state.projectId = payload;
    },
    setClinicalTrial: (state, { payload }) => {
      state.clinicalTrial = payload;
    },
    setSponsor: (state, { payload }) => {
      state.sponsor = payload;
    },
    setDescription: (state, { payload }) => {
      state.description = payload;
    },
    setStandards: (state, { payload }) => {
      state.standards = payload;
    },
    setMeasurement: (state, { payload }) => {
      state.measurement = payload;
    },
    setSegmentId: (state, { payload }) => {
      state.segmentId = payload;
    },
    setProjectSettings: (state, { payload }) => {
      state.projectId = payload.projectId;
      state.segmentId = payload.segmentId;
      state.clinicalTrial = payload.clinicalTrial;
      state.sponsor = payload.sponsor;
      state.description = payload.description;
      state.measurement = payload.measurement;
      state.standards = payload.standards;
    },
    clearSettings: (state) => {
      state.clinicalTrial = '';
      state.sponsor = '';
      state.description = '';
      state.measurement = 'Metric';
      state.standards = '';
      state.projectId = '';
      state.segmentId = [];
    },
  },
});

export const {
  setProjects,
  setProjectId,
  setProjectSettings,
  setClinicalTrial,
  setDescription,
  setSponsor,
  setMeasurement,
  setStandards,
  setSegmentId,
  clearSettings,
} = projectSlice.actions;

export default projectSlice.reducer;
