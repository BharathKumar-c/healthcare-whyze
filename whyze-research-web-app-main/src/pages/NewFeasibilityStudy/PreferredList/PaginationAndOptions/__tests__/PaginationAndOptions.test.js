import { fireEvent, render } from '@testing-library/react';
import PaginationAndOptions from '..';
import React from 'react';
import Store from '../../../../../redux/store';
import { Provider } from 'react-redux';
import { act } from 'react-dom/test-utils';
import { setSegments } from '../../../../../redux/reducer/feasibilityStudyReducer';

function renderWithContext(element) {
  const { container } = render(<Provider store={Store}>{element}</Provider>);
  return { container, Store };
}
const demoSites = [
  {
    id: 3,
    isChecked: true,
    isLinked: false,
    location: 'London, UK',
    noOfConnectedSites: 87,
    noOfNotConnectedSites: 6,
    star: '4.1',
    university: 'University College Hospital at Westmoreland Street',
  },
];
describe('testcases for PaginationAndOptions', () => {
  test('create snapshots for PaginationAndOptions', () => {
    const { container } = renderWithContext(<PaginationAndOptions />);
    act(() => {
      const { segments, activeSegment } = JSON.parse(
        JSON.stringify(Store.getState().feasibilityStudyReducer),
      );
      segments[activeSegment.index].feasibilityStudy.push(demoSites);
      Store.dispatch(setSegments(segments));
    });
    expect(container).toMatchSnapshot();
  });
  test('test deletefunction', () => {
    const { container } = renderWithContext(<PaginationAndOptions />);
    const deleteFunction = container.getElementsByClassName(
      'preferred-list-option-alignment',
    );
    fireEvent.click(deleteFunction[0].firstChild);
    expect(deleteFunction[0].firstChild.textContent).toBe('Delete.svgDelete');
  });
  test('selected  preferredSites tab ', () => {
    const { container } = renderWithContext(
      <PaginationAndOptions tab={'preferredSites'} noOfSelectedSites={1} />,
    );

    const preferredSitescheckbox = container.querySelector(
      'input[name="sites"]',
    );

    const OptionAlignment = container.getElementsByClassName(
      'preferred-list-option-alignment',
    );

    fireEvent.click(preferredSitescheckbox);
    fireEvent.click(OptionAlignment[1].firstChild);
    const preferredListText = container.getElementsByClassName(
      'preferred-list-text',
    );

    expect(preferredListText[2].firstChild.textContent).toBe('Mark initiated');
  });
  test('selected contacted  tab ', () => {
    const { container } = renderWithContext(
      <PaginationAndOptions tab={'contacted'} noOfSelectedSites={1} />,
    );

    const OptionAlignment = container.getElementsByClassName(
      'preferred-list-option-alignment',
    );

    fireEvent.click(OptionAlignment[2].firstChild);
    fireEvent.click(OptionAlignment[1].firstChild);
    const preferredListText = container.getElementsByClassName(
      'preferred-list-text',
    );

    expect(preferredListText[2].firstChild.textContent).toBe('Mark initiated');
  });

  test('pagination arrows trigger test', () => {
    const { container } = renderWithContext(
      <PaginationAndOptions tab={'initiated'} noOfSelectedSites={1} />,
    );
    const OptionAlignment = container.getElementsByClassName(
      'preferred-list-arrow-icon',
    );

    fireEvent.click(OptionAlignment[0].firstChild);
    fireEvent.click(
      container.getElementsByClassName('preferred-list-arrow-right')[0]
        .firstChild,
    );
    expect(OptionAlignment[0].firstChild).toBeInTheDocument();
  });
  test('current page is notEqual to 1 or more', () => {
    const { container } = renderWithContext(
      <PaginationAndOptions tab={'initiated'} currentPage={2} />,
    );
    const OptionAlignment = container.getElementsByClassName(
      'preferred-list-arrow-icon',
    );

    fireEvent.click(OptionAlignment[0].firstChild);
    fireEvent.click(
      container.getElementsByClassName('preferred-list-arrow-right')[0]
        .firstChild,
    );
    expect(OptionAlignment[0].firstChild).toBeInTheDocument();
  });
  test('current page not equal to totalpage', () => {
    const { container } = renderWithContext(
      <PaginationAndOptions tab={'initiated'} currentPage={1} totalPage={2} />,
    );
    const OptionAlignment = container.getElementsByClassName(
      'preferred-list-arrow-icon',
    );

    fireEvent.click(OptionAlignment[0].firstChild);
    fireEvent.click(
      container.getElementsByClassName('preferred-list-arrow-right')[0]
        .firstChild,
    );
    expect(OptionAlignment[0].firstChild).toBeInTheDocument();
  });
});
