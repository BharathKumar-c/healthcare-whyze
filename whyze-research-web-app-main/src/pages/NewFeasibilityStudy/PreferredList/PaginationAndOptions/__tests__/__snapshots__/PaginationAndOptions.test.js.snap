// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testcases for PaginationAndOptions create snapshots for PaginationAndOptions 1`] = `
<div>
  <div
    class="preferred-list-pagination-container"
  >
    <div
      class="preferred-list-alignment"
    >
      <div
        class="preferred-list-alignment"
      >
        <div
          class="preferred-list-checkbox"
        >
          <label
            class="ant-checkbox-wrapper"
          >
            <span
              class="ant-checkbox"
            >
              <input
                class="ant-checkbox-input"
                name="sites"
                type="checkbox"
                value=""
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
          </label>
          <span
            class="ant-typography"
            style="margin-left: 1rem;"
          >
            0 Sites
          </span>
        </div>
        <div
          class="ant-divider ant-divider-vertical preferred-list-divider"
          role="separator"
        />
      </div>
    </div>
    <div
      class="preferred-list-options"
    >
      <div
        class="preferred-list-option-alignment"
      >
        <div
          class="preferred-list-option"
        >
          <div
            class="preferred-list-icon"
          >
            <svg>
              Delete.svg
            </svg>
          </div>
          <span
            class="ant-typography preferred-list-text"
          >
            Delete
          </span>
        </div>
      </div>
      <div
        class="preferred-list-option-alignment"
      >
        <div
          class="preferred-list-option"
        >
          <div
            class="preferred-list-icon"
          >
            <svg>
              Contact.svg
            </svg>
          </div>
          <span
            class="ant-typography preferred-list-text"
          >
            Mark contacted
          </span>
        </div>
      </div>
      <div
        class="preferred-list-option-alignment"
      >
        <div
          class="preferred-list-option"
        >
          <div
            class="preferred-list-icon"
          >
            <svg>
              Initiated.svg
            </svg>
          </div>
          <span
            class="ant-typography preferred-list-text"
          >
            Mark initiated
          </span>
        </div>
      </div>
    </div>
    <div
      class="preferred-list-alignment"
    >
      <span
        class="ant-typography preferred-list-pagination-text"
      >
        0-0 of 0
      </span>
      <div
        class="preferred-list-arrow-icon"
      >
        <svg
          class="preferred-list-arrow-left"
        >
          LeftArrow.svg
        </svg>
        <svg
          class="preferred-list-arrow-right"
        >
          RightArrow.svg
        </svg>
      </div>
    </div>
  </div>
</div>
`;
